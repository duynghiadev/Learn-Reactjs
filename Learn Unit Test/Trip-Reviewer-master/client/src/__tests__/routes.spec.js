import React from 'react';
import { create } from 'react-test-renderer';

import Routes from '../routes';

jest.mock('./../hoc/layout', () => 'Layout');
jest.mock('react-router-dom', () => ({
  Switch: 'Switch',
  Route: 'Route',
}));

const mockComponent = props => <Routes {...props} />;

describe('<Routes />', () => {
  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
