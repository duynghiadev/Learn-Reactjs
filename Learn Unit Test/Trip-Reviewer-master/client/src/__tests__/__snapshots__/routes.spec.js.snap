// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> should render component 1`] = `
<Layout>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/reset-password"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/user/logout"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/user/register"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/user"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/user/user-reviews"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/user/edit-profile"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/user/edit-post/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/trip/add"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/trips/:id"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
</Layout>
`;
